server/src/middlewares.rs

```diff
@@ -72,3 +73,20 @@ impl<S> Middleware<S> for ErrorHandlers {
         }
     }
 }
+
+/// Read request before returning response. This is not required for RFC compliance, however:
+///
+/// * a lot of clients are not able to deal with unread request payloads
+/// * in HTTP over unix domain sockets, not reading the request payload might cause the client's
+///   write() to block forever due to a filled up socket buffer
+pub struct ReadRequestMiddleware;
+
+impl<S> Middleware<S> for ReadRequestMiddleware {
+    fn response(&self, req: &HttpRequest<S>, resp: HttpResponse) -> Result<Response, Error> {
+        let consume_request = req.payload().for_each(|_| Ok(()));
```

Comment:
what happens if someone else already read the payload earlier. Does this silently do nothing?

Reply:
I didn't look into how actix works here but all tests should fail otherwise.